src/v/security/mtls.cc

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "security/mtls.h"
+
+#include <boost/algorithm/string/case_conv.hpp>
+#include <fmt/format.h>
+#include <fmt/ostream.h>
+#include <fmt/ranges.h>
+
+#include <regex>
+#include <stdexcept>
+
+namespace security::tls {
+
+namespace detail {
+
+static constexpr const char* const rule_pattern{
```

Comment:
Perhaps the leading "RULE:" is not needed?  The rule is going to either be something with a placeholder in it (user input) or "default", so we can distinguish without RULE:

Reply:
Existing practice: https://github.com/apache/kafka/blob/trunk/clients/src/test/java/org/apache/kafka/common/security/ssl/SslPrincipalMapperTest.java